ambry-cloud/src/main/java/com.github.ambry.cloud/azure/AzureCloudDestination.java

```diff
@@ -68,11 +70,18 @@
 
   private static final Logger logger = LoggerFactory.getLogger(AzureCloudDestination.class);
   private static final String THRESHOLD_PARAM = "@threshold";
+  private static final String LIMIT_PARAM = "@limit";
+  private static final String TIME_SINCE_PARAM = "@timesince";
   private static final String BATCH_ID_QUERY_TEMPLATE = "SELECT * FROM c WHERE c.id IN (%s)";
   static final String DEAD_BLOBS_QUERY_TEMPLATE =
-      "SELECT * FROM c WHERE (c." + CloudBlobMetadata.FIELD_DELETION_TIME + " BETWEEN 1 AND " + THRESHOLD_PARAM + ")"
-          + " OR (c." + CloudBlobMetadata.FIELD_EXPIRATION_TIME + " BETWEEN 1 AND " + THRESHOLD_PARAM + ")";
+      "SELECT TOP " + LIMIT_PARAM + " * FROM c WHERE (c." + CloudBlobMetadata.FIELD_DELETION_TIME + " BETWEEN 1 AND " + THRESHOLD_PARAM + ")"
+          + " OR (c." + CloudBlobMetadata.FIELD_EXPIRATION_TIME + " BETWEEN 1 AND " + THRESHOLD_PARAM + ")"
+          + " ORDER BY c." + CloudBlobMetadata.FIELD_UPLOAD_TIME + " ASC";
+  static final String ENTRIES_SINCE_QUERY_TEMPLATE =
```

Comment:
If multiple blobs have the same time stamp and we do the same query multiple times, do blobs' order always same ?

`findEntriesSince()` skips blobs before `latestBlob`, if `latestBlob`'s position is non-deterministic, some blobs will be skipped incorrectly. 

Can we order by both timestamp and id?

Reply:
Another conner case is, in a very short time window but lots of blobs with same timestamp, after `findEntriesSince()`first query, new blob with same timestamp comes, if it has smaller alphabeta order than `latestBlob`, it will be skipped. (assumed we ordered by both time and id)

This looks like a race condition if we do bi-directional data synchronization.

Reply:
I tried adding id to the order by clause but unfortunately CosmosDB doesn't support that:
`The order by query does not have a corresponding composite index that it can be served from.`
However, I changed the logic to only skip the lastBlobId in the token while iterating, not all the blobs *until* that one.  So nothing should be skipped now.
For what it's worth, the chance of an uploadTime collision for the same partition is extremely unlikely since the upload latency is over 100ms.

Reply:
Cool. 
But let's add some comment for this unlikely scenario, in case we want to scale it, for example, multiple VCRs upload same partition.

Reply:
Added.